Issues in linking

satacker@ubuntu:~/Desktop/Chat-App-Boost.Asio-$ make
-- Configuring done
-- Generating done
-- Build files have been written to: /home/satacker/Desktop/Chat-App-Boost.Asio-
Scanning dependencies of target app
[ 25%] Building CXX object CMakeFiles/app.dir/main.cpp.o
[ 50%] Building CXX object CMakeFiles/app.dir/session/session.cpp.o
[ 75%] Building CXX object CMakeFiles/app.dir/server/server.cpp.o
[100%] Linking CXX executable app
/usr/bin/ld: CMakeFiles/app.dir/main.cpp.o: in function `boost::asio::detail::posix_event::posix_event()':
main.cpp:(.text._ZN5boost4asio6detail11posix_eventC2Ev[_ZN5boost4asio6detail11posix_eventC5Ev]+0x51): undefined reference to `pthread_condattr_setclock'
/usr/bin/ld: CMakeFiles/app.dir/main.cpp.o: in function `boost::asio::detail::posix_thread::~posix_thread()':
main.cpp:(.text._ZN5boost4asio6detail12posix_threadD2Ev[_ZN5boost4asio6detail12posix_threadD5Ev]+0x2a): undefined reference to `pthread_detach'
/usr/bin/ld: CMakeFiles/app.dir/main.cpp.o: in function `boost::asio::detail::posix_thread::join()':
main.cpp:(.text._ZN5boost4asio6detail12posix_thread4joinEv[_ZN5boost4asio6detail12posix_thread4joinEv]+0x2f): undefined reference to `pthread_join'
/usr/bin/ld: CMakeFiles/app.dir/main.cpp.o: in function `boost::asio::detail::posix_thread::start_thread(boost::asio::detail::posix_thread::func_base*)':
main.cpp:(.text._ZN5boost4asio6detail12posix_thread12start_threadEPNS2_9func_baseE[_ZN5boost4asio6detail12posix_thread12start_threadEPNS2_9func_baseE]+0x3e): undefined reference to `pthread_create'
/usr/bin/ld: CMakeFiles/app.dir/main.cpp.o: in function `boost::asio::detail::posix_signal_blocker::posix_signal_blocker()':
main.cpp:(.text._ZN5boost4asio6detail20posix_signal_blockerC2Ev[_ZN5boost4asio6detail20posix_signal_blockerC5Ev]+0x68): undefined reference to `pthread_sigmask'
/usr/bin/ld: CMakeFiles/app.dir/main.cpp.o: in function `boost::asio::detail::posix_signal_blocker::~posix_signal_blocker()':
main.cpp:(.text._ZN5boost4asio6detail20posix_signal_blockerD2Ev[_ZN5boost4asio6detail20posix_signal_blockerD5Ev]+0x31): undefined reference to `pthread_sigmask'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/app.dir/build.make:133: app] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/app.dir/all] Error 2
make: *** [Makefile:103: all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,5 +13,5 @@ IF(Boost_FOUND)
 ENDIF(Boost_FOUND)
 
 SET(USED_LIBS ${Boost_SYSTEM_LIBRARY} ${Boost_THREAD_LIBRARY} ${Boost_REGEX_LIBRARY})
-
-add_executable(server main.cpp session/session.cpp session/session.h server/server.cpp server/server.h)
+link_libraries(pthread)
+add_executable("${PROJECT_NAME}-app" main.cpp session/session.cpp session/session.h server/server.cpp server/server.h)

